src/devices/cpu/mipsx/mipsxdasm.cpp

```diff
@@ -0,0 +1,588 @@
+			{
+				util::stream_format(stream, "hsc");
+			}
+			else
+			{
+				util::stream_format(stream, "illegal hsc form");
+			}
+			break;
+		}
+
+		case 2:
+		{
+			// movtos - Move to Special Register
+			// movtos rSrc1, SpecialReg
+
+			// psw  001
+			// md   010
+			// pcm1 100
+			u16 comp = get_compute_compfunc(opcode);
+			if ((comp & 0xffe) == 0x000)
+			{
+				u8 dest = get_src2_dest(opcode);
+
+				if (dest == 0x00)
+				{
+					u8 src = get_src1(opcode);
+					util::stream_format(stream, "movtos %s, %01x", get_regname(src), comp & 0x7);
+				}
+				else
+				{
+					util::stream_format(stream, "illegal movtos form");
+				}
+			}
+			else
+			{
+				util::stream_format(stream, "illegal movtos form");
+			}
+			break;
+
+		}
+
+		case 3:
+		{
+			// movfrs - Move from Special Register
+			// movfrs SpecialReg, rDest
+
+			// psw  001
+			// md   010
+			// pcm4 100
+			u16 comp = get_compute_compfunc(opcode);
+			if ((comp & 0xffe) == 0x000)
+			{
+				u8 src = get_src1(opcode);
+				if (src == 0x00)
+				{
+					u8 dest = get_src2_dest(opcode);
+					util::stream_format(stream, "movfrs %01x, %s", comp & 0x7, get_regname(dest));
+				}
+				else
+				{
+					util::stream_format(stream, "illegal movfrs form");
+				}
+			}
+			else
+			{
+				util::stream_format(stream, "illegal movfrs form");
+			}
+			break;
+
+		}
+
+		case 4:
+		{
+			int imm17 = get_imm17(opcode);
+			imm17 = util::sext(imm17 & 0x1ffff, 17);
+			util::stream_format(stream, "addi %s, %s, %08x", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), imm17);
+			break;
+		}
+
+		case 5:
+		{
+			u16 comp = get_compute_compfunc(opcode);
+			if (comp == 0x03)
+			{
+				u8 dest = get_src2_dest(opcode);
+				u8 src = get_src1(opcode);
+
+				if ((src == 0x00) && (dest == 0x00))
+				{
+					util::stream_format(stream, "jpc");
+				}
+				else
+				{
+					util::stream_format(stream, "illegal jpc form");
+				}
+			}
+			else
+			{
+				util::stream_format(stream, "illegal jpc form");
+			}
+			break;
+		}
+
+		case 6:
+		{
+			if ((opcode & 0x07fff807) == 0x00000003)
+			{
+				int vector = (opcode & 0x000007f8) >> 3;
+				util::stream_format(stream, "trap %02x", vector);
+			}
+			else
+			{
+				util::stream_format(stream, "illegal trap form");
+			}
+			break;
+		}
+
+		case 7:
+		{
+			u16 comp = get_compute_compfunc(opcode);
+			if (comp == 0x03)
+			{
+				u8 dest = get_src2_dest(opcode);
+				u8 src = get_src1(opcode);
+
+				if ((src == 0x00) && (dest == 0x00))
+				{
+					util::stream_format(stream, "jpcrs");
+				}
+				else
+				{
+					util::stream_format(stream, "illegal jpcrs form");
+				}
+			}
+			else
+			{
+				util::stream_format(stream, "illegal jpcrs form");
+			}
+			break;
+		}
+
+		default:
+		{
+			util::stream_format(stream, "Unhandled TY3 (%08x)", opcode);
+			break;
+		}
+		}
+		break;
+	}
+
+	default:
+	{
+		util::stream_format(stream, "Unhandled (%08x)", opcode);
+		break;
+	}
+	}
+
+	return 4 | SUPPORTED;
```

Comment:
Does not appear that any flags are in fact supported?

Reply:
probably not, I'm not 100% clear what SUPPORTED is meant to mean in this context anyway, so it was just copy+pasted from the core I used as a base.

Reply:
SUPPORTED means that you produce the STEP_OVER/COND/OUT flags when appropriate, and thus the debugger can provide the step over functionality.